### PR TITLE
Topoplotseries

### PR DESCRIPTION
**Pull request by vladdez**
- continuous and categorical nrows 
- refactoring of eeg_series and topo_series
- issue #188 
- issue #120
- issue #198 
- [adding topoplot_axis to give more control over individual topoplots](https://github.com/unfoldtoolbox/UnfoldMakie.jl/pull/186/commits/ab9010344d78a385152c4d09d279cb922cd28700)
- more docs

### CODE REVIEW

**Comment by vladdez** [comment]
The level of change is huge, while faceting is still far away
Maybe accept this part? and think on faceting in next PR?

File: docs/src/helper.md [hunk 1]
```diff
@@ -5,6 +5,6 @@ UnfoldMakie.config_kwargs!
 UnfoldMakie.eeg_matrix_to_dataframe
 UnfoldMakie.rel_to_abs_bbox
 UnfoldMakie.to_positions
-UnfoldMakie.df_timebin
+UnfoldMakie.df_grouping
```
Comment (behinger) [inline docs/src/helper.md:8]: df_binning ?

File: src/eeg_series.jl [hunk 2]
```diff
@@ -285,15 +240,16 @@ function interctive_toposeries(interactive_scatter, single_topoplot)
     end
 end
 
-function create_axis_options(xlim_topo, ylim_topo)
+function create_axis_options()
```
Comment (behinger) [inline src/eeg_series.jl:243]: I guess all of these could in principle live in plotConfig?

File: src/plot_topoplotseries.jl [hunk 3]
```diff
@@ -32,13 +31,16 @@ Multiple miniature topoplots in regular distances.
 - `positions::Vector{Point{2, Float32}} = nothing`\\
     Specify channel positions. Requires the list of x and y positions for all unique electrode.
 - `interactive_scatter = nothing`\\
-    Enable interactive mode. \\ 
+    Enable interactive mode.\\ 
     If you create `obs_tuple = Observable((0, 0, 0))` and pass it into `interactive_scatter` you can change observable indecies by clicking topopplot markers.\\
     `(0, 0, 0)` corresponds to the indecies of row of topoplot layout, column of topoplot layout and channell. 
+- `topoplot_axes = (;)`\\
```
Comment (behinger) [inline src/plot_topoplotseries.jl:37]:
```suggestion
- `topoplot_axes::NamedTuple = (;)`\\
```

File: src/plot_topoplotseries.jl [hunk 4]
```diff
@@ -32,13 +31,16 @@ Multiple miniature topoplots in regular distances.
 - `positions::Vector{Point{2, Float32}} = nothing`\\
     Specify channel positions. Requires the list of x and y positions for all unique electrode.
 - `interactive_scatter = nothing`\\
-    Enable interactive mode. \\ 
+    Enable interactive mode.\\ 
     If you create `obs_tuple = Observable((0, 0, 0))` and pass it into `interactive_scatter` you can change observable indecies by clicking topopplot markers.\\
```
Comment (behinger) [inline src/plot_topoplotseries.jl:35]:
```suggestion
    If you create `obs_tuple = Observable((0, 0, 0))` and pass it into `interactive_scatter` you can update the observable tuple with the indices of the clicked topopplot markers (row / column / electrode) .\\
```

File: src/plot_topoplotseries.jl [hunk 5]
```diff
@@ -8,12 +8,11 @@ Multiple miniature topoplots in regular distances.
...
     `Figure`, `GridLayout`, `GridPosition`, or GridLayoutBase.GridSubposition to draw the plot.
 - `data::Union{<:Observable{<:DataFrame},DataFrame}`\\
-    DataFrame with data or Observable DataFrame. Requires a `time` column, but could be also specified in mapping.x. 
+    DataFrame with data or Observable DataFrame. Requires a `time` column, but could be also specified in mapping.x by any continous or categorical value. 
 
 ## Keyword arguments (kwargs)
 - `bin_width::Real = nothing`\\
-    Number specifing the width of time bin.\\
-    `bin_width` is in units of the `data.time` column.\\
+    Number specifing the width of bin of continous x-value in its units.\\
```
Comment (behinger) [inline src/plot_topoplotseries.jl:15]:
```suggestion
    Number specifing the width of bin of continuous x-value in its units.\\
```

File: src/plot_topoplotseries.jl [hunk 6]
```diff
@@ -8,12 +8,11 @@ Multiple miniature topoplots in regular distances.
 - `f::Union{GridPosition, GridLayout, GridLayoutBase.GridSubposition, Figure}`\\
     `Figure`, `GridLayout`, `GridPosition`, or GridLayoutBase.GridSubposition to draw the plot.
 - `data::Union{<:Observable{<:DataFrame},DataFrame}`\\
-    DataFrame with data or Observable DataFrame. Requires a `time` column, but could be also specified in mapping.x. 
+    DataFrame with data or Observable DataFrame. Requires a `time` column, but could be also specified in mapping.x by any continous or categorical value.
```
Comment (behinger) [inline src/plot_topoplotseries.jl:11]:
```suggestion
    DataFrame with data or Observable DataFrame. By default requires a `time` column, but can be overwritten by specifying `mapping=(; x=:my_column)` with any continuous or categorical column. 
```

File: src/plot_topoplotseries.jl [hunk 7]
```diff
@@ -32,13 +31,16 @@ Multiple miniature topoplots in regular distances.
...
-    Enable interactive mode. \\ 
+    Enable interactive mode.\\ 
     If you create `obs_tuple = Observable((0, 0, 0))` and pass it into `interactive_scatter` you can change observable indecies by clicking topopplot markers.\\
     `(0, 0, 0)` corresponds to the indecies of row of topoplot layout, column of topoplot layout and channell. 
+- `topoplot_axes = (;)`\\
+    Here you can flexibly change configurations of topoplots.\\
+    To see all options just type `?Axis` in REPL.
 - `mapping.x = :time`\\
-    Specification of x value, could be any contionous variable. 
+    Specify x-value. Can be any continuous variable.
```
Comment (behinger) [inline src/plot_topoplotseries.jl:41]: this is a bit missleading, you should rather describe the `mapping = (;)` relationship, right?

Comment (behinger) [reply src/plot_topoplotseries.jl:41]: or you could make a section "## facet options"

**Review by behinger — APPROVED** [review]

**Comment by behinger** [comment]
nice! only some small comments

File: src/eeg_series.jl (hunk 2 above)
Comment (vladdez) [reply src/eeg_series.jl:243]: I also had this though, i just was not sure how to implement that

**Pull request merged**